kadi/commands/commands_v2.py

```diff
@@ -1362,6 +1367,32 @@ def get_matching_block_idx(cmds_arch, cmds_recent):
     return idx0_arch, idx0_recent
 
 
+def get_list_for_matching(cmds: CommandTable) -> list[tuple]:
```

Comment:
Seems like there should be a new unit test for this function / bugfix.

Reply:
This is covered in the functional testing. The issue with adapting that functional test into a unit test is that it requires a snapshot of how the commands archive looked on 2024-Feb-04. This is not possible to construct from the current flight archive because the Not-run commands have all be removed.

In theory the test could check for the presence of this 100 Mb file in $SKA/data/kadi and run if available. Let me know if you are not satisfied with the one-time functional test. My thinking was that given the code documentation it seems not credible that future updates will break this particular fix.

Reply:
OK, when I wrote my comment I was thinking it would not be that bad to generate short cmds archive test data to show that OBS statement is correctly updated by new processing. But if you think that doesn't have much value that's fine.